tests/api-watch-diff/test.ml

```diff
@@ -1,13 +1,53 @@
+open Api_watch_diff
+
+let pp_diff_list fmt diffs =
+  let pp_item_change fmt = function
+    | Added -> Format.fprintf fmt "Added"
+    | Removed -> Format.fprintf fmt "Removed"
+    | Modified _ -> Format.fprintf fmt "Modified"
+  in
+  let pp_diff fmt = function
+    | Value (name, change) ->
+        Format.fprintf fmt "Value (%s, %a)" name pp_item_change change
+    | Any -> Format.fprintf fmt "Any"
+  in
+  Format.fprintf fmt "[%a]" (Format.pp_print_list pp_diff) diffs
+
 let%expect_test "test_diff_interface" =
   let result = Api_watch_diff.diff_interface ~reference:[] ~current:[] in
-  Format.printf "%b" result;
-  [%expect {|false|}]
+  Format.printf "%a" pp_diff_list result;
+  [%expect {|[]|}]
+
+let x_int_value_signature =
+  Test_helpers.compile_interface {|
+    val x : int
+    |}
+
+let x_string_value_signature =
+  Test_helpers.compile_interface {|
+    val x : string
+    |}
+
+let%expect_test "Simple value, identical" =
+  let result =
+    Api_watch_diff.diff_interface ~reference:x_int_value_signature
+      ~current:x_int_value_signature
+  in
+  Format.printf "%a" pp_diff_list result;
+  [%expect {|[]|}]
```

Comment:
Even I was thinking about that. We can also remove the comments now as they feel redundant with the new signature definitions!

Reply:
@NathanReb Should I start with the new PR for this?

Reply:
Sure! That's not urgent though it can also be done iteratively as we modify parts of the test suite.

It would be nice to do it here for the tests that are added or modified by this PR, such as this one, the simple test modif one and the type + value modif one.